packages/protocol/contracts/governance/Governance.sol

```diff
@@ -657,20 +662,30 @@ contract Governance is
   function revokeVotes() external nonReentrant returns (bool) {
     address account = getAccounts().voteSignerToAccount(msg.sender);
     Voter storage voter = voters[account];
-    for (uint256 i = 0; i < dequeued.length; i = i.add(1)) {
-      VoteRecord storage voteRecord = voter.referendumVotes[i];
-      (Proposals.Proposal storage proposal, Proposals.Stage stage) =
-        requireDequeuedAndDeleteExpired(
-          voteRecord.proposalId,
-          i
-        );
+    for (
+      uint256 dequeueIndex = 0;
+      dequeueIndex < dequeued.length;
+      dequeueIndex = dequeueIndex.add(1)
+    ) {
+      VoteRecord storage voteRecord = voter.referendumVotes[dequeueIndex];
+      (Proposals.Proposal storage proposal, Proposals.Stage stage) = requireDequeuedAndDeleteExpired(
+        voteRecord.proposalId,
+        dequeueIndex
+      );
       require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
-      proposal.updateVote(voteRecord.weight, 0, voteRecord.value, Proposals.VoteValue.None);
-      proposal.networkWeight = proposal.networkWeight - voteRecord.weight;
-      voter.referendumVotes[i] = VoteRecord(Proposals.VoteValue.None, voteRecord.proposalId, 0);
+      uint256 weight = voteRecord.weight;
```

Comment:
I don't feel super strongly about this but I think this variable unnecessarily expands the memory stack
we might want to just reference via `voteRecord.weight` throughout

Reply:
same for `voteRecord.value`

Reply:
Yeah, even I thought so but for some reason I had to do this. I will revert this back. thanks again.